Userland/Utilities/pkg/AvailablePort.cpp

```diff
@@ -28,11 +28,11 @@
 #include <Shell/PosixParser.h>
 #include <Shell/Shell.h>
 
-static bool is_installed(HashMap<String, InstalledPort>& installed_ports_database, StringView package_name)
-{
-    auto port = installed_ports_database.find(package_name);
-    return port != installed_ports_database.end();
-}
+// static bool is_installed(HashMap<String, InstalledPort>& installed_ports_database, StringView package_name)
+// {
+//     auto port = installed_ports_database.find(package_name);
+//     return port != installed_ports_database.end();
+// }
```

Comment:
Don't leave commented-out code in-tree. That said, this looks like it should still be used?

Reply:
> That said, this looks like it should still be used?

No I am using this definition in the function. So double code is not required.

Reply:
But what reason is there to _not_ use the function? It's already there, it's quite literally more effort to remove it than to just leave it as-is.

Reply:
Ah, I missed that you are using `installed_port` afterwards. In that case, not using the function is probably fine, although the point about not leaving in commented-out code still stands and we could think about making this function `Optional<InstalledPort> get_installed_port(...)` instead.

Reply:
I have better plans for pkg, we can discuss in discord. Basically making a library out of it so that in case anyone wants to build GUI, they can also use same set of APIs as we have for console. What do you think?

Reply:
>  Basically making a library out of it so that in case anyone wants to build GUI, they can also use same set of APIs as we have for console. What do you think?

This would be better served as moving the entire `pkg` infrastructure into `Applications/Package` and having two entry points: `pkg` for the command line tool, and `pkg --gui` or some such as the GUI tool. I can't imagine we would need more than one GUI Package manager in the base system.

Reply:
Yes, I have this in plan. Once this PR is tested. I would start working on Application instead of utility.

Reply:
Moved to discord `#package-manager` channel